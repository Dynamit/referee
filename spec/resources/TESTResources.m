/** THIS FILE IS AUTOGENERATED. DO NOT EDIT. CHANGES WILL BE DESTROYED. */

#import "TESTResources.h"

/** UIStoryboard accessors. */
@implementation __TESTStoryboards

- (UIStoryboard *)Main
{
	return [UIStoryboard storyboardWithName:@"Main" bundle:[NSBundle mainBundle]];
}
- (UIStoryboard *)LaunchScreen
{
	return [UIStoryboard storyboardWithName:@"LaunchScreen" bundle:[NSBundle mainBundle]];
}

@end

/** UITableViewCell identifier accessors. */
@implementation __TESTTableViewCells

- (NSString *)Cell
{
	return @"Cell";
}

@end

/** UICollectionViewCell identifier accessors. */
@implementation __TESTCollectionViewCells


@end

/** UIViewController accessors. */
@implementation __TESTViewControllers

- (UINavigationController *)newNavigationController1
{
	return (UINavigationController *)[[UIStoryboard storyboardWithName:@"Main" bundle:[NSBundle mainBundle]] instantiateViewControllerWithIdentifier:@"NavigationController1"];
}
- (DetailViewController *)newDetailViewController
{
	return (DetailViewController *)[[UIStoryboard storyboardWithName:@"Main" bundle:[NSBundle mainBundle]] instantiateViewControllerWithIdentifier:@"DetailViewController"];
}
- (UISplitViewController *)newSplitViewController
{
	return (UISplitViewController *)[[UIStoryboard storyboardWithName:@"Main" bundle:[NSBundle mainBundle]] instantiateViewControllerWithIdentifier:@"SplitViewController"];
}
- (UIViewController *)newGenericController
{
	return (UIViewController *)[[UIStoryboard storyboardWithName:@"Main" bundle:[NSBundle mainBundle]] instantiateViewControllerWithIdentifier:@"GenericController"];
}
- (MasterViewController *)newMasterViewController
{
	return (MasterViewController *)[[UIStoryboard storyboardWithName:@"Main" bundle:[NSBundle mainBundle]] instantiateViewControllerWithIdentifier:@"MasterViewController"];
}
- (UINavigationController *)newNavigationController2
{
	return (UINavigationController *)[[UIStoryboard storyboardWithName:@"Main" bundle:[NSBundle mainBundle]] instantiateViewControllerWithIdentifier:@"NavigationController2"];
}

@end

/** Segue identifier accessors. */
@implementation __TESTSegues

- (NSString *)showDetail
{
	return @"showDetail";
}

@end

/** Main resource reference class. */
@implementation __TESTResources

- (instancetype)init
{
    self = [super init];

    if (self) {
        self.storyboards = [[__TESTStoryboards alloc] init];
        self.tableCells = [[__TESTTableViewCells alloc] init];
        self.collectionCells = [[__TESTCollectionViewCells alloc] init];
        self.viewControllers = [[__TESTViewControllers alloc] init];
        self.segues = [[__TESTSegues alloc] init];
    }

    return self;
}

+ (instancetype)sharedInstance
{
    static id _self = nil;
    static dispatch_once_t once_Token;
    dispatch_once(&once_Token, ^{
      _self = [[super alloc] init];
    });
    return _self;
}

@end
