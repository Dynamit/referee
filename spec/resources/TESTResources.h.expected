/** THIS FILE IS AUTOGENERATED. DO NOT EDIT. CHANGES WILL BE DESTROYED. */

@import UIKit;

/** ATTENTION: Use this macro to access resources. */
#define TESTResources ([__TESTResources sharedInstance])

@interface __TESTStoryboards : NSObject

- (UIStoryboard *)Main;
- (UIStoryboard *)LaunchScreen;

@end

@interface __TESTTableViewCells : NSObject

- (NSString *)Cell;

@end

@interface __TESTCollectionViewCells : NSObject


@end

@interface __TESTViewControllers : NSObject

- (UINavigationController *)NavigationController1;
- (DetailViewController *)DetailViewController;
- (UISplitViewController *)SplitViewController;
- (UIViewController *)GenericController;
- (MasterViewController *)MasterViewController;
- (UINavigationController *)NavigationController2;

@end

@interface __TESTSegues : NSObject

- (NSString *)showDetail;

@end

/** Global resource reference accessor class. */
@interface __TESTResources : NSObject

/** Singleton accessor. */
+ (instancetype)sharedInstance;

/** Resource accessors. */
@property (nonatomic, strong) __TESTStoryboards *storyboards;
@property (nonatomic, strong) __TESTTableViewCells *tableCells;
@property (nonatomic, strong) __TESTCollectionViewCells *collectionCells;
@property (nonatomic, strong) __TESTViewControllers *viewControllers;
@property (nonatomic, strong) __TESTSegues *segues;

@end
